File: 2895-minimum-processing-time/2895-minimum-processing-time.java
class Solution {
    public int minProcessingTime(List<Integer> pro, List<Integer> tasks) {
        Collections.sort(tasks);
        Collections.sort(pro, Collections.reverseOrder());
        int ans=Integer.MIN_VALUE;
        int j=0;
        for(int i=0;i<pro.size();i++){
            for(int k=1;k<=4;k++){
                ans=Math.max(ans,pro.get(i)+tasks.get(j));
                j++;
            }
        }
        return ans;
    }
}

File: 2895-minimum-processing-time/2895-minimum-processing-time.cpp
class Solution {
public:
    int minProcessingTime(vector<int>& t, vector<int>& v) {
        int n=v.size();
        sort(t.begin(),t.end());
        sort(v.begin(),v.end());
        int j=n-1;
        int m=t.size();
        int ans=0;
        for(int i=0;i<m;i++)
        {
            int c=0;
            while(c<4)
            {
                ans=max(ans,t[i]+v[j]);
                c++;
                j--;
            }
        }
        return ans;
    }
};